pypeit/core/coadd.py

```diff
@@ -549,24 +552,46 @@ def solve_poly_ratio(wave, flux, ivar, flux_ref, ivar_ref, norder, mask = None,
     # Determine an initial guess
     ratio = robust_median_ratio(flux, ivar, flux_ref, ivar_ref, mask=mask, mask_ref=mask_ref,
                                 ref_percentile=ref_percentile, max_factor=scale_max)
+    wave_min = wave.min()
+    wave_max = wave.max()
+
     if 'poly' in model:
         guess = np.append(ratio, np.zeros(norder-1))
+        scale_fun = lambda x: x
+        err_scale_fun = lambda x: x
+        mask_fun = lambda x: np.ones_like(x)
     elif 'square' in model:
         guess = np.append(np.sqrt(ratio), np.zeros(norder-1))
+        scale_fun = np.sqrt
+        err_scale_fun = lambda x: 2 * np.sqrt(x)
+        mask_fun = lambda x: x >= 0
     elif 'exp' in model:
         guess = np.append(np.log(ratio), np.zeros(norder-1))
+        scale_fun = np.log
+        err_scale_fun = lambda x: x
+        mask_fun = lambda x: x > 0
     else:
         msgs.error('Unrecognized model type')
 
-    wave_min = wave.min()
-    wave_max = wave.max()
 
     # Now compute median filtered versions of the spectra which we will actually operate on for the fitting. Note
     # that rejection will however work on the non-filtered spectra.
     med_width = (2.0*np.ceil(median_frac/2.0*nspec) + 1).astype(int)
     flux_med, ivar_med = median_filt_spec(flux, ivar, mask, med_width)
     flux_ref_med, ivar_ref_med = median_filt_spec(flux_ref, ivar_ref, mask_ref, med_width)
 
+    # Here we compute a linear regression to the scaled flux in order to get a better guess.
+    if norder > 1:
+            # we need scale_mask to mask negative points in the polyfit
+            scale_mask = mask_fun(flux_ref_med)
+
```

Comment:
Doing this with robust_fit now, take a look at my branch.

Reply:
your solution in mosfire_oct20 looks good